### ISSUE TITLE
Upgrade to Vue 3

### ISSUE THREAD

**Issue by walterdeboer** [DependencyTrack/frontend#446]
### Current Behavior

Frontend now uses outdated Vue 2

### Proposed Behavior

Move to Vue 3 after #445 is resolved:
- upgrage Vue
- Upgrade to CoreUI 4.x (still alpha release now)
- Replace vue2-org-tree by https://www.npmjs.com/package/vue3-blocks-tree
- upgrade other dependencies

### Checklist

- [X] I have read and understand the [contributing guidelines](https://github.com/DependencyTrack/dependency-track/blob/master/CONTRIBUTING.md#filing-issues)
- [X] I have checked the [existing issues](https://github.com/DependencyTrack/frontend/issues) for whether this enhancement was already requested

**Comment by sekwah41** [comment]
I will open a draft pr and start work on a new attempt at this. Originally I was considering moving to Vite though since the last time I looked vue has suggested switching to vite and moved the old methods to maintenance mode.

**Comment by stohrendorf** [comment]
@nscuro IMHO, an in-place migration is difficult. Some stuff doesn't have a Vue 3 equivalent as far as I know, and other components would need a careful review and rewrite with changed functionality, but to be honest, I didn't look too much into migrating stuff, since I am focusing on covering current behaviour only, so that a migration wouldn't break too many things. My gut feeling is that a migration to Vue 3 has the potential to break quite a lot of stuff; I agree with the opinions posted before. As a disclaimer, I am *not* a frontend developer, but it *feels* like a lot of things are legacy stuff that's hard to get around. I am using AI roughly 20% of the time to get a starting point for tests, but the majority of the work is to actually get the tests passing, since there's a lot of tightly coupled things going on that the AI doesn't know about.

TLDR, it's a lot of work necessary, but it's not requiring a full re-write. Some parts need to be re-written, sure, because some 3rd party components need different replacements, but compared to the whole complexity, it's a minor thing.

**Comment by stohrendorf** [comment]
@OrbisK I am not familiar with nuxtl, but your idea sounds like it needs a full rewrite, since you suggested a different framework. because I am not a frontend dev, I am curious whether there are pratical benefits to your suggestion, or if it's just personal preference. for example, I have a personal preference for vue and cypress, but since I am not a frontend dev, I can't present valuable facts or insights. but I am pretty sure that a change of framework would still require full tests so that it ensures a re-write doesn't change behaviour. your plan for a rewrite suggests that a lot of things would be broken for some time.

**Comment by OrbisK** [comment]
Nuxt is simply a meta framework built on top of Vue. The main benefit, in my opinion, is that it provides an opinion on the basics, such as routing and middlewares. This makes it easier to develop and contribute.

All pages/components will still be Vue components. 

But you are right, that's 100% personal preference.

**Comment by nscuro** [comment]
@OrbisK Sorry for getting back to you so late. Thanks so much for having a look at this.

> I think it will take around 200 hours to rewrite the front end completely, plus some extra time for testing and management.

Ooof, that's a lot.

> If there are no objections, I will then sketch out the structure roughly and write down any sub-tasks.

That would be absolutely fantastic!

**Comment by OrbisK** [comment]
Progress:

<img width="1910" height="946" alt="Image" src="https://github.com/user-attachments/assets/707fc9fc-6b70-44af-9c2d-8257a209b103" />

@nscuro I am missing test data. Could you share some "good" SBOM that I can use to have some real vulnerabilities?

Next steps:

1. finish dashboard as example page
2. include i18n
3. auth middleware and permissions

After that, we can divide up the work if anyone wants to help  😅

**Comment by OrbisK** [comment]
I am still working on it. I currently have to complete some exams until march.
